src/smk/layer/layer-vector.js

```diff
@@ -21,54 +21,82 @@ include.module( 'layer.layer-vector-js', [ 'layer.layer-js' ], function () {
         var cv = $( '<canvas width="' + width * mult + '" height="' + height + '">' ).get( 0 )
         var ctx = cv.getContext( '2d' )
 
-        var styles = [].concat( self.config.style )
+        const legendData = [];
+        legendData.push({
+            title: self.config.legend.title || self.config.title,
+            style: self.config.style
+        });
+
+        if (self.config.conditionalStyles) {
+            self.config.conditionalStyles.forEach(cs => {
+                const defaultStyle = self.config.style ? self.config.style : {};
+                self.config.conditionalStyles.forEach(conditionalStyle => {
+                    conditionalStyle.conditions.forEach(condition => {
+                        const combinedStyle = Object.assign({}, defaultStyle);
+                        Object.assign(combinedStyle, condition.style);
+                        legendData.push({
+                            title: condition.label || condition.value,
+                            style: combinedStyle,
+                            indent: true
+                        });
+                    });
+                });
+            });
+        }
 
         return SMK.UTIL.resolved( 0 )
             .then( drawPoint )
             .then( drawLine )
             .then( drawFill )
             .then( function () {
-                return [ {
-                    url: cv.toDataURL( 'image/png' ),
-                    title: self.config.legend.title || self.config.title
-                } ]
+                return legendData;
             } )
 
         function drawPoint( offset ) {
             if ( !self.config.legend.point ) return offset 
 
-            return SMK.UTIL.makePromise( function ( res, rej ) {
-                if ( styles[ 0 ].markerUrl ) {
-                    var img = $( '<img>' )
-                        .on( 'load', function () {
-                            var r = img.width / img.height
-                            if ( r > 1 ) r = 1 / r
-                            ctx.drawImage( img, offset, 0, height * r, height )
-                            res( offset + width )
-                        } )
-                        .on( 'error', res )
-                        .attr( 'src', viewer.resolveAttachmentUrl( styles[ 0 ].markerUrl, null, 'png' ) )
-                        .get( 0 )
-                }
-                else {
+            if (legendData.length === 1 && legendData[0].style.markerUrl) {
+                loadMarkerImage(offset)
+                .then((offset) => {
+                    legendData[0].url = cv.toDataURL( 'image/png' );
+                    return offset;
+                });
+            } else {
+                legendData.forEach(leg => {
+                    const st = leg.style;
```

Comment:
Point of personal preference, but variable names that match the intended meaning make code more readable for others. eg `const legendStyle = legendItem.style;`

No need to change, just a thought for future consideration.

Reply:
I went ahead and renamed these variables for clarity.